win32ss/user/ntuser/caret.c

```diff
@@ -29,17 +31,21 @@ co_IntDrawCaret(PWND pWnd, PTHRDCARETINFO CaretInfo)
        return;
     }
 
-    hdc = UserGetDCEx(pWnd, 0, DCX_USESTYLE | DCX_WINDOW);
+    hdc = UserGetDCEx(pWnd, NULL, DCX_USESTYLE);
     if (!hdc)
     {
         ERR("GetDC failed\n");
         return;
     }
 
-    if (pWnd->hrgnUpdate)
-    {
-       NtGdiSaveDC(hdc);
-    }
+    NtGdiSaveDC(hdc);
+
+    IntGetClientRect(pWnd, &rcClient);
+    NtGdiIntersectClipRect(hdc,
```

Comment:
After removing DCX_WINDOW this should not be required anymore. On the other hand, when I look at the implementation of UserGetDCEx, it seems like it doesn't set the client region as clip region. So to me it looks like the bug is there. Can you investigate that?

Reply:
See https://jira.reactos.org/browse/CORE-10201 .
I confirmed that GetDC and GetDCEx(hwnd, NULL, DCX_CACHE) enable clipping.